qiime/core/type/signature.py

```diff
@@ -94,10 +110,30 @@ def defaults(self):
             (name, spec.default) for name, spec in self.parameters.items()
             if spec.has_default()])
 
-    def _parse_signature(self, callable, inputs, parameters, outputs):
+    @property
+    def descriptions(self):
+        params = {**self.inputs, **self.parameters, **self.outputs}
```

Comment:
This was a helper being utilized [here](https://github.com/qiime2/q2cli/pull/108/files#diff-8b6223dfee196a9888f17c22ddabcf94R269) in `q2cli` in the same manner as `defaults`.